pkg/operator/ceph/cluster/watcher.go

```diff
@@ -130,6 +147,290 @@ func (c *clientCluster) onK8sNode(ctx context.Context, object runtime.Object) bo
+		return pkgerror.Wrapf(err, "failed to list watchers for pool/imageName %s/%s.", rbdPV.Spec.CSI.VolumeAttributes["pool"], rbdPV.Spec.CSI.VolumeAttributes["imageName"])
+	}
+
+	logger.Infof("rbd status to get the client ips %v", string(buf))
+	ips, err := rbdStatusUnMarshal(buf)
+	if err != nil {
+		return pkgerror.Wrapf(err, "failed to unmarshal rbd status output")
+	}
+	if len(ips) != 0 {
+		err = c.createNetworkFence(ctx, rbdPV, node, cluster, ips)
+		if err != nil {
+			return pkgerror.Wrapf(err, "failed to create network fence for node %q", node.Name)
+		}
+	}
+
+	return nil
+}
+
+func rbdStatusUnMarshal(output []byte) ([]string, error) {
+	type rbdStatus struct {
+		Watchers []struct {
+			Address string `json:"address"`
+		} `json:"watchers"`
+	}
+
+	var rbdStatusObj rbdStatus
+	err := json.Unmarshal([]byte(output), &rbdStatusObj)
+	if err != nil {
+		return []string{}, pkgerror.Wrapf(err, "failed to unmarshal rbd status output")
+	}
+
+	watcherIPlist := []string{}
+	for _, watcher := range rbdStatusObj.Watchers {
+		// split with separation '/' to remove nounce and concatenating `/32` to define a network with only one IP address
+		address := strings.Split(watcher.Address, "/")[0] + "/32"
+		watcherIPlist = append(watcherIPlist, address)
+	}
+	return watcherIPlist, nil
+}
+
+func (c *clientCluster) createNetworkFence(ctx context.Context, pv corev1.PersistentVolume, node *corev1.Node, cluster *cephv1.CephCluster, cidr []string) error {
+	logger.Warningf("Blocking node IP %s", cidr)
+
+	secretName := pv.Annotations["volume.kubernetes.io/provisioner-deletion-secret-name"]
+	secretNameSpace := pv.Annotations["volume.kubernetes.io/provisioner-deletion-secret-namespace"]
+	if secretName == "" || secretNameSpace == "" {
+		storageClass, err := c.context.Clientset.StorageV1().StorageClasses().Get(ctx, pv.Spec.StorageClassName, metav1.GetOptions{})
+		if err != nil {
+			return pkgerror.Wrap(err, "failed to get storage class to fence volume")
+		}
+		secretName = storageClass.Parameters["csi.storage.k8s.io/provisioner-secret-name"]
+		secretNameSpace = storageClass.Parameters["csi.storage.k8s.io/provisioner-secret-namespace"]
+	}
+
+	networkFence := &addonsv1alpha1.NetworkFence{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      node.Name,
+			Namespace: cluster.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(cluster, cephv1.SchemeGroupVersion.WithKind("CephCluster")),
+			},
+		},
+		Spec: addonsv1alpha1.NetworkFenceSpec{
+			Driver:     pv.Spec.CSI.Driver,
+			FenceState: addonsv1alpha1.Fenced,
+			Secret: addonsv1alpha1.SecretSpec{
+				Name:      secretName,
+				Namespace: secretNameSpace,
+			},
+			Cidrs: cidr,
+			Parameters: map[string]string{
+				"clusterID": pv.Spec.CSI.VolumeAttributes["clusterID"],
+			},
+		},
+	}
+
+	err := c.client.Create(ctx, networkFence)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	attempt := 60
+	sleep := 1 * time.Second
+	for attempt > 0 {
```

Comment:
For polling, try the `utilwait.PollImmediate()` used other places in rook code.

Reply:
Actually, why does the operator need to wait for the fenced status? Why not create the CR and then continue reconciling immediately?

Reply:
hmm, since you mentioned now I agree, since creation of this is handle by csiaddons and not rook  so no point in keep checking

Reply:
removed the check